packages/itwin/ec3-widget/src/components/ReportTableSelector.tsx

```diff
@@ -0,0 +1,119 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { IModelApp } from "@itwin/core-frontend";
+import { ComboBox, Label, toaster } from "@itwin/itwinui-react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import useValidator, { NAME_REQUIREMENTS } from "../hooks/useValidator";
+import type { Configuration } from "./Template";
+import "./ReportTableSelector.scss";
+import { useODataClient } from "./api/context/ODataClientContext";
+
+export interface ReportTableSelectorProps {
+  selectedReportTable: string;
+  template: Configuration;
+  placeHolder: string;
+  onChange: (table: string, numCols: string[], strCols: string[]) => Promise<void>;
+  setLoading: (val: boolean) => void;
+}
+
+export const ReportTableSelector = ({
+  selectedReportTable,
+  template,
+  placeHolder,
+  onChange,
+  setLoading,
+}: ReportTableSelectorProps) => {
+  const [validator, _showValidationMessage] = useValidator();
+  const [reportTable, setReportTable] = useState(selectedReportTable);
+  const [reportTables, setReportTables] = useState<string[] | undefined>(undefined);
+  const reportingClientApi = useODataClient();
+
+  const updateData = useCallback(async (reportTableName: string) => {
+    if (!IModelApp.authorizationClient)
```

Comment:
Try not to use `IModelApp` statics inside components. It would be best to pass down a `getAccessToken` function as a prop.

Reply:
Or have it in a context.

Reply:
made context for getting access token